api/eth_types.go

```diff
@@ -83,6 +83,9 @@ func (e EthBytes) MarshalJSON() ([]byte, error) {
 	if len(e) == 0 {
 		return json.Marshal("0x")
 	}
+	if len(e) == 1 && e[0] == 0 {
+		return json.Marshal("0x0")
+	}
 	s := hex.EncodeToString(e)
 	if len(s)%2 == 1 {
 		s = "0" + s
```

Comment:
This is probably superfluous. My tests show that `hex.EncodeToString` always returns 2 * n characters, and I expect the behaviour to be consistent with [`EncodeLen`](https://pkg.go.dev/encoding/hex#EncodedLen), which also returns 2 * n as per asserted behavior in docs.